src/ui/components/URLBar/URLBar.tsx

```diff
@@ -119,35 +121,41 @@ export function URLBar() {
       <RenderArea
         name="navigation-bar"
         children={(children) => {
-          let text: React.ReactNode;
+          let maybeText: React.ReactNode;
           // This check is done to work around an unavoidable inconsistent state
           // where this callback function is called first because of pathname change
           // and then later because some <Content /> element was added
           let automaticTitle: null | string = null;
           if (!children.length && pathnameRef.current !== pathname) {
-            text = null;
+            maybeText = null;
           } else {
             if (children.length) {
-              text = children;
+              // we want to wrap url content only in case it is a string
+              // in this case it is enough to get only the first element and check its type later
+              maybeText = children[0];
```

Comment:
Aren't we losing elements?

Reply:
We shouldn't cause this case will work if only the first item of children is text. There shouldn't be a case with text + component